docs/ingress/virtualserver.md

```diff
@@ -0,0 +1,325 @@
+
+In VSCode, open the */manifsts/brewz/virtaul-server.yml* file and add an ErrorPage resource; example below.
+
+```yaml
+---
+apiVersion: k8s.nginx.org/v1
+kind: VirtualServer
+metadata:
+  name: brewz
+spec:
+  host: brewz.f5demo.com
+  upstreams:
+    - name: spa
+      service: spa
+      port: 80
+    - name: api
+      service: api
+      port: 8000
+      healthCheck:
+        enable: true
+        path: /api/stats
+        interval: 20s
+        jitter: 3s
+        port: 8000
+  routes:
+    - path: /
+      action:
+        pass: spa
+    - path: /api
+      action:
+        pass: api
+    - path: /images
+      action:
+        proxy:
+          upstream: api
+          rewritePath: /images
+```
+
+Run the following command on the K3s server via the UDF *SSH* or *Web Shell* Access Methods to test that our API services is still up and has a health health check:
+```bash
+# Find the Brewz Access Method's Host
+HOST=`curl -s metadata.udf/deployment | jq '.deployment.components[] | select(.name == "k3s") | .accessMethods.https[] | select(.label == "Brewz") | .host' -r`
+curl -k https://$HOST/api/products/123
+```
+
+# ErrorPage
+While the Brews developers were able to break their monolith application into microservies their APIs are not always returning a JSON response.  A good example is when you lookup a product that does not exist.  The API returns a 400 HTTP response code but the body payload is *""Could not find the product!"*.
+
+Run the following command on the K3s server via the UDF *SSH* or *Web Shell* Access Methods to test this output:
+```bash
+# Find the Brewz Access Method's Host
+HOST=`curl -s metadata.udf/deployment | jq '.deployment.components[] | select(.name == "k3s") | .accessMethods.https[] | select(.label == "Brewz") | .host' -r`
+curl -k https://$HOST/api/products/1234
+```
+
+Ideally, the development team will fix this issue in the API code but we can also help by performing a quick fix via our VirtualServer configuration.
+
+In VSCode, open the */manifsts/brewz/virtaul-server.yml* file and add an ErrorPage resource; example below.
+
+```yaml
+apiVersion: k8s.nginx.org/v1
+kind: VirtualServer
+metadata:
+  name: brewz
+spec:
+  host: brewz.f5demo.com
+  upstreams:
+    - name: spa
+      service: spa
+      port: 80
+    - name: api
+      service: api
+      port: 8000
+      healthCheck:
+        enable: true
+        path: /api/stats
+        interval: 20s
+        jitter: 3s
+        port: 8000
+  routes:
+    - path: /
+      action:
+        pass: spa
+    - path: /api
+      action:
+        pass: api
+      errorPages:
+        - codes: [404]
+          return:
+            code: 404
+            type: application/json
+            body: |
+              {\"msg\": \"Could not find the product!\"}
+            headers:
+            - name: x-debug-original-statues
```

Comment:
Not familiar with this header, but is there a typo in it?

Reply:
fixed